com.unity.multiplayer.mlapi/Runtime/Connection/PendingClient.cs

```diff
@@ -8,22 +8,11 @@ public class PendingClient
         /// <summary>
         /// The ClientId of the client
         /// </summary>
-        public ulong ClientId;
-
+        public ulong ClientId { get; internal set; }
+        
         /// <summary>
-        /// The state of the connection process for the client
+        /// Whether or not the client has sent a connection request
         /// </summary>
-        public State ConnectionState;
-
-        /// <summary>
-        /// The states of a connection
-        /// </summary>
-        public enum State
-        {
-            /// <summary>
-            /// Client is in the process of doing the connection handshake
-            /// </summary>
-            PendingConnection
-        }
+        public bool HasSentConnectionRequest { get; internal set; }
```

Comment:
rather than removing `ConnectionState` property and `State` enum from the public API, we might basically introduce another enum element called `SentConnectionRequest` and that way we wouldn't cause public API changes but still satisfy our extra state requirement here, right? what do you think?

Reply:
I disagree, because this fix is not about a state change. The state is changed when the connection request is processed. This is in between that, it simply fixes multiple connection requests.

Reply:
I also think the State should have been removed earlier, when all the other states were removed. But since they weren't I did it in this commit.

I could put it back, but it's redudant since there is only one state.

Reply:
how `HasSentConnectionRequest` is not a state? :)

Reply:
Perhaps it is.

I will convert it to the state enum.

But I still think the ConnectionState should be public get internal set. So it's still breaking.

Same reason as the ClientId change.